include/ck_tile/ops/permute/kernel/generic_permute_kernel.hpp

```diff
@@ -40,10 +40,11 @@ struct GenericPermute
 {
     using Problem = ck_tile::remove_cvref_t<Problem_>;
 
-    using DataType                      = remove_cvref_t<typename Problem::DataType>;
-    static constexpr index_t kBlockSize = Problem::kBlockSize;
-    static constexpr index_t kMaxRanks  = Problem::kMaxRanks;
-    static constexpr bool KeepLastDim   = Problem::KeepLastDim;
+    using DataType                           = remove_cvref_t<typename Problem::DataType>;
+    static constexpr index_t kBlockSize      = Problem::kBlockSize;
+    static constexpr index_t KernelBlockSize = Problem::kBlockSize;
```

Comment:
I think we should use this opportunity to refactor and unify the usage of block size constants. Right now this form results in two constants visible here and there `kBlockSize` and `KernelBlockSize` - this makes more confusion than help. 
Please resort to just single constant which should be derived from operator pipeline problem (or other structure if no pipeline problem) definition.

Reply:
Personally I'd prefer just BlockSize without the `k` suffix ;) But what about the `WorkgroupSize` ? It's more natural for HIP

Reply:
ok, I will replace it with BlockSize, it is closer with existing code.